lib/GLPI/Agent/SNMP/MibSupport/LinuxAppliance.pm

```diff
@@ -257,6 +268,52 @@ sub run {
 
     my $firmware;
     if ($manufacturer eq 'Synology') {
+        
+        my $diskModels = $self->walk(dsmDiskModel)
+        or return;
+
+        my $diskNames = $self->walk(dsmDiskName)
+        or return;
+
+        my $volumesNames = $self->walk(dsmRaidName)
+        or return;
+
+        my $volumesFreeSizes = $self->walk(dsmRaidFreeSize)
+        or return;
+
+        my $volumesTotalSizes = $self->walk(dsmRaidTotalSize)
+        or return;
```

Comment:
You should always indent before `or` if used this way.
```suggestion
        my $diskModels = $self->walk(dsmDiskModel)
            or return;

        my $diskNames = $self->walk(dsmDiskName)
            or return;

        my $volumesNames = $self->walk(dsmRaidName)
            or return;

        my $volumesFreeSizes = $self->walk(dsmRaidFreeSize)
            or return;

        my $volumesTotalSizes = $self->walk(dsmRaidTotalSize)
            or return;
```
But I'm not agree with a point: if no disk or no volume is found, all the `or return` conditions can lead to break the Synology inventory.
It would be better to test if a walk part is defined just before you want to use it. If it is not defined, you just need to skip the related part of the inventory.